index.js

```diff
@@ -1,14 +1,29 @@
 const EventEmitter = require('events')
+const mutexify = require('mutexify/promise')
 
 module.exports = class ReadyResource extends EventEmitter {
-  constructor () {
+  constructor ({ suspended = false } = {}) {
     super()
 
     this.opening = null
     this.closing = null
 
     this.opened = false
     this.closed = false
+
+    // State machine:
```

Comment:
I can remove this if you prefer (added because the code is not trivial, so this way the intent of the pattern is clear)